src/common/storage/backends/s3/Client.js

```diff
@@ -0,0 +1,205 @@
+                    config: this.config,
+                    bucketName: bucketName,
+                    httpMethod: httpMethod,
+                    path: path,
+                })
+            });
+        } catch (err) {
+            await this.throwError('Get PreSigned URL', err);
+        }
+        return await res.json();
+    };
+
+    S3Storage.prototype._stat = async function (path) {
+        let res;
+        try {
+            res = await this.fetch('/statObject', {
+                method: 'post',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    config: this.config,
+                    bucketName: this.bucketName,
+                    path: path
+                })
+            });
+        } catch (err) {
+            await this.throwError('Stat Object', err);
+        }
+
+        return await res.json();
+    };
+
+    S3Storage.prototype.putFile = async function (path, content) {
+        await this._createBucketIfNeeded();
+        const httpInfo = await this._getPreSignedURL(this.bucketName, 'put', path);
+        let res;
+        try {
+            res = await this.fetch(httpInfo.queryURL, {
+                method: httpInfo.httpMethod.toUpperCase(),
+                body: content
+            });
+        } catch (err) {
+            this.throwError('Put File', err);
+        }
+        this.logger.debug(`Successfully uploaded the file to ${httpInfo.queryURL}, server response ${res.body}`);
+        const metadata = await this._stat(path);
+        metadata.bucketName = this.bucketName;
+        metadata.path = path;
+        return this.createDataInfo(metadata);
+    };
+
+    S3Storage.prototype.getFile = async function (dataInfo) {
+        const downloadURL = await this.getDownloadURL(dataInfo);
+        let resObj;
+        try {
+            resObj = await this.fetch(downloadURL);
+        } catch (err) {
+            this.throwError('Get File', err);
+        }
+        this.logger.debug('Successfully downloaded artifact from S3 Storage');
+        return require.isBrowser ? await resObj.arrayBuffer() : Buffer.from(await resObj.arrayBuffer());
+    };
+
+    S3Storage.prototype.deleteFile = async function (dataInfo) {
+        const {data} = dataInfo;
+        const {queryURL} = await this._getPreSignedURL(data.bucketName, 'DELETE', data.path);
+        try {
+            return await this.fetch(queryURL, {method: 'DELETE'});
+        } catch (err) {
+            this.throwError('Delete File', err);
+        }
+    };
+
+
+    S3Storage.prototype.getDownloadURL = async function (dataInfo) {
+        const {data} = dataInfo;
+        const res = await this._getPreSignedURL(data.bucketName, 'GET', data.path);
+        return res.queryURL;
+    };
+
+    S3Storage.prototype.getMetadata = async function (dataInfo) {
+        const metaData = {size: dataInfo.data.size};
```

Comment:
metadata is actually one word so the camel case form would still be `metadata`

Reply:
It now just returns an inline variable.